File: R/tests/testthat/test-reference-runner.R
#!/usr/bin/env Rscript
# Script: test-reference-runner.R
# Objective: Verify the shared R reference runner command-line interface.
# Author: Yi Yu
# Created: 2026-06-24
# Last updated: 2026-06-24
# Inputs: R reference runner script.
# Outputs: Testthat assertions.
# Usage: Loaded by testthat::test_local("R", filter = "reference-runner").
# Dependencies: testthat

test_that("reference runner accepts both case identifiers", {
  script <- testthat::test_path(
    "..",
    "..",
    "..",
    "examples",
    "R",
    "run_reference_case.R"
  )
  if (!file.exists(script)) {
    skip("The shared R reference runner is available only from a source checkout.")
  }
  help <- system2("Rscript", c(script, "--help"), stdout = TRUE)

  expect_true(any(grepl("--case", help, fixed = TRUE)))
  expect_true(any(grepl("nsw_griffith_heavy_imazapic", help, fixed = TRUE)))
  expect_true(any(grepl("sa_minnipa_heavy_imazapic", help, fixed = TRUE)))
})

File: python/tests/test_reference_runner.py
#!/usr/bin/env python3
"""
Script: test_reference_runner.py
Objective: Verify the shared Python reference runner command-line workflow.
Author: Yi Yu
Created: 2026-06-24
Last updated: 2026-06-24
Inputs: Shared case inputs under examples/data.
Outputs: Pytest assertions for runner artifacts.
Usage: python -m pytest python/tests/test_reference_runner.py -q
Dependencies: os, pathlib, subprocess, sys
"""

from __future__ import annotations

import os
from pathlib import Path
import subprocess
import sys


ROOT = Path(__file__).parents[2]


def test_python_runner_writes_expected_schema(tmp_path: Path) -> None:
    env = os.environ.copy()
    env["MPLBACKEND"] = "Agg"
    subprocess.run(
        [
            sys.executable,
            str(ROOT / "examples" / "python" / "run_reference_case.py"),
            "--case",
            "nsw_griffith_heavy_imazapic",
            "--input-dir",
            str(ROOT / "examples" / "data" / "nsw_griffith_heavy_imazapic"),
            "--output-dir",
            str(tmp_path),
        ],
        check=True,
        env=env,
    )
    expected = {
        "bulk_density.csv",
        "climate_forcing.csv",
        "fit_diagnostics.csv",
        "fit_parameters.csv",
        "metadata.json",
        "objective_profiles.csv",
        "observations_prepared.csv",
        "predictions.csv",
    }
    assert expected <= {path.name for path in tmp_path.iterdir()}
